cmd/lncli/cmd_pay.go

```diff
@@ -46,6 +46,13 @@ var (
 			"<hex_value>,.. For example: --data 3438382=0a21ff. " +
 			"Custom record ids start from 65536.",
 	}
+
+	maxHtlcsFlag = cli.UintFlag{
+		Name: "max_htlcs",
```

Comment:
Perhaps we should call this something similar to `max_shards`? Otherwise, users my confuse this with the existing max HTLC value used in routing policies.

Reply:
Yes, there could be confusion, although this is plural, not `max_htlc`. Also on the `Payment` message, there is an array called `htlcs`. Introducing a new term for the same thing may introduce a different type of confusion. 

I prefer to keep naming consistent over confusion with the max htlc value. Another option would be to rename `Payment.htlcs` to `Payment.shards`, but that would be a (compile time) breaking change.

Reply:
Do you suggest to rename to `shards` everywhere?

Reply:
I don't think it would be a problem using both names. Shards are the pieces a payment is split into, while HTLCs are all the attempted routes. A payment can obviously have many many HTLCs in its list, but shards will always be at most this number.

Reply:
`max_splits` could also be another option here.

Reply:
That is a good point @halseth. So then a shard would be defined as a settled htlc. 

`max_splits`, nice find, but isn't that `max_shards-1`?